addons/mail/models/ir_attachment.py

```diff
@@ -71,5 +82,90 @@ def _attachment_format(self, commands=False):
                     'res_id': attachment.res_id,
                     'res_model': attachment.res_model,
                 })
-            res_list.append(res)
-        return res_list
+            attachments.append(res)
+        return attachments
+
+    @api.model
+    def _get_link_preview_from_url(self, url):
+        try:
+            page = requests.get(url, timeout=1)
+        except requests.exceptions.RequestException:
+            return False
+
+        if page.status_code != requests.codes.ok:
+            return False
+
+        image_mimetype = [
+            'image/bmp',
+            'image/gif',
+            'image/jpeg',
+            'image/png',
+            'image/tiff',
+            'image/x-icon',
+        ]
+        if page.headers['Content-Type'] in image_mimetype:
+            return self._get_link_preview_from_url_image(url)
+        if 'text/html' in page.headers['Content-Type']:
+            return self._get_link_preview_from_url_html(url, page.content)
+        return False
+
+    def _get_link_preview_from_url_image(self, url):
+        image = self._fetch_image(url)
+        data = {
+            'url': url,
+            'name': url,
+            'description': False,
+            'raw': image,
+            'mimetype': 'image/o-linkpreview-image',
+        }
+        return data
+
+    def _fetch_image(self, image_url):
+        request_image = False
+        try:
+            request_image = requests.get(image_url, timeout=1)
+        except requests.exceptions.RequestException:
+            return False
+        if request_image and request_image.status_code != requests.codes.ok:
+            return False
+        return image_process(
+            request_image.content,
+            size=(300, 300),
+            verify_resolution=True
+        )
+
+    def _get_link_preview_from_url_html(self, url, content):
```

Comment:
confusing name, it doesn't get anything from `url`, the `url` parameter here is merely to fill some values